src/Middleware/RequestThrottling/src/RequestThrottlingMiddleware.cs

```diff
@@ -35,6 +35,10 @@ public RequestThrottlingMiddleware(RequestDelegate next, ILoggerFactory loggerFa
             {
                 throw new ArgumentException("The value of 'options.MaxConcurrentRequests' must be specified.", nameof(options));
             }
+            if (_requestThrottlingOptions.MaxConcurrentRequests < 0)
+            {
+                throw new ArgumentException("The value of 'options.MaxConcurrentRequests' must be a positive integer.", nameof(options));
+            }
```

Comment:
I feel this is the wrong place to do validation. The setter on the property should throw for invalid values:

https://github.com/aspnet/AspNetCore/blob/master/src/Middleware/RequestThrottling/src/RequestThrottlingOptions.cs#L19

Reply:
Note we don't currently use that pattern in this middleware (or any other middleware).

Recommendation: take this change as is and then follow up with a change to move this all to Options Validation.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-2.2#options-validation